mopidy/core/playlists.py

```diff
@@ -203,16 +207,27 @@ def refresh(self, uri_scheme=None):
         :param uri_scheme: limit to the backend matching the URI scheme
         :type uri_scheme: string
         """
-        if uri_scheme is None:
-            futures = [b.playlists.refresh()
-                       for b in self.backends.with_playlists.values()]
-            pykka.get_all(futures)
+        futures = {}
+        backends = {}
+        playlists_loaded = False
+
+        for backend_scheme, backend in self.backends.with_playlists.items():
+            backends.setdefault(backend, set()).add(backend_scheme)
+
+        for backend, backend_schemes in backends.items():
+            if uri_scheme is None or uri_scheme in backend_schemes:
+                futures[backend] = backend.playlists.refresh()
```

Comment:
I see the same issue here as with `library.refresh()`, where we do the same call n times and use the last result that is available.